firebase-firestore/src/main/java/com/google/firebase/firestore/local/LocalStore.java

```diff
@@ -381,17 +382,18 @@ public SnapshotVersion getLastRemoteSnapshotVersion() {
             MaybeDocument doc = entry.getValue();
             MaybeDocument existingDoc = existingDocs.get(key);
 
-            // If a document update isn't authoritative, make sure we don't
-            // apply an old document version to the remote cache. We make an
-            // exception for SnapshotVersion.MIN which can happen for
-            // manufactured events (e.g. in the case of a limbo document
-            // resolution failing).
+            // If a document update isn't authoritative, make sure we don't apply an old document
+            // version to the remote cache. We make an exception for SnapshotVersion.MIN which can
+            // happen for manufactured events (e.g. in the case of a limbo document resolution
+            // failing).
```

Comment:
As mentioned on chat, this comment is now misleading.  It says we don't write old document versions to the remote cache except we make an exception for SnapshotVersion.MIN (implying we _do_ write it to cache).  With your change, this is no longer true.

Perhaps remove "We make an exception ..." from this comment and add a new comment inside your new else-if case that says something like:

This is a manufactured event (since version==SnapshotVersion.NONE), likely as a result of a failed limbo resolution lookup.  Rather than write the NoDocument to cache with an invalid version, we just remove whatever might currently be in cache.

Reply:
I updated it just as you wrote this comment:) Let me know if the new version is a better description. Thanks!

Reply:
Haha.  Nice!  LGTM.